### PR TITLE
WIP: Setup unit testing for react project closes #21

### LINKED ISSUE #21: Setup frontend unit testing
**User Story**
As a frontend developer, I'd like to setup unit testing for react components, so that failures can be caught earlier in the SDLC and manual testing efforts are reduced.

**Acceptance Criteria**

- Tests created for existing components
- Tests grouped into a `tests` folder or similar
- Tests run by `yarn test`
- Research into snapshot tests, implement if appropriate

**Notes**
Good example here https://github.com/rpaschoal/unit-testing-on-the-front-end-101-workshop, also many good examples online.

Will probably use *Jest* and *Enzyme*

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**

N/A

**Describe the solution you'd like**

N/A

**Describe alternatives you've considered**

N/A

### PR DESCRIPTION
**Pull request by jordansimsmith**
GitHub Issue (If applicable): #21 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

- No unit testing for react components

## What is the new behavior?

- Tests run on `yarn test`
- Example tests written for dummy Hello components
- Jest and Enzyme dependencies setup

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -2,14 +2,6 @@
...
   "private": true,
-  "dependencies": {
-    "@testing-library/jest-dom": "^4.2.4",
-    "@testing-library/react": "^9.3.2",
-    "@testing-library/user-event": "^7.1.2",
-    "react": "^16.13.0",
-    "react-dom": "^16.13.0",
-    "react-scripts": "3.4.0"
-  },
   "scripts": {
```
Comment (Kalashnikkov) [inline client/package.json]: @jordansimsmith inside here. Don't know why my comment didn't link to here :(

**Comment by Matteas-Eden** [comment]
How do we test this to ensure it's setup correctly?

**Comment by jordansimsmith** [comment]
> Does this need to include a script for `yarn test` (as described) to run `jest` instead of `react-scripts test` ?

React scripts test will handle jest and other configuration. Running `yarn test` works as is.

**Comment by jordansimsmith** [comment]
> How do we test this to ensure it's setup correctly?

Checkout the branch and run `yarn test`. Should interactively run all tests (2) and should pass. You can try adding your own test and see that it picks it up if you really want to test it :)

File: client/package.json [hunk 2]
```diff
@@ -2,14 +2,6 @@
   "name": "client",
   "version": "0.1.0",
   "private": true,
-  "dependencies": {
-    "@testing-library/jest-dom": "^4.2.4",
```
Comment (PreetPatel) [inline client/package.json]: Do we want to lock dependencies?

File: client/package.json [hunk 3]
```diff
@@ -30,5 +22,18 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "dependencies": {
+    "react": "^16.13.0",
```
Comment (PreetPatel) [inline client/package.json]: Again would be better to lock down a version since react keeps changing so much

Comment (jordansimsmith) [reply client/package.json]: The ^ locks down to minor version changes, I think that would be acceptable for this project. Breaking changes would more likely occur over major version changes.

i.e. ^16.13.0 is valid from 16.13.0 to  < 17.0.0

File: client/package.json (hunk 2 above)
Comment (jordansimsmith) [reply client/package.json]: Same as below

File: client/package.json (hunk 3 above)
Comment (PreetPatel) [reply client/package.json]: Depends on if the packages we use utilise the convention that a non breaking change is only introduced in minor versions. What I have noticed however with packages that are not backed by a large community, they introduce breaking changes with a 0.x.0 version. Again if its a critical issue, npm will warn us if the dev has configured it properly or deprecated it

**Pull request merged**